tests/test_scatterwidgets.py

```diff
@@ -1,58 +1,404 @@
-from typing import Any
-
 import numpy as np
-import pandas as pd
+import pyqtgraph as pg
+import pytest
 from napari_spatialdata._model import DataModel
-from napari_spatialdata._scatterwidgets import MatplotlibWidget
+from napari_spatialdata._scatterwidgets import PlotWidget
+from pyqtgraph.Qt import QtCore, QtGui, QtWidgets
+from shapely.geometry import Polygon
+
+DATA_LEN = 100
+
+
+def normalize_coordinates(coords):
+    return [tuple(map(float, coord)) for coord in coords]
+
+
+def coordinates_are_equal(coords1, coords2, tol=1e-6):
+    norm_coords1 = normalize_coordinates(coords1)
+    norm_coords2 = normalize_coordinates(coords2)
+
+    if len(norm_coords1) != len(norm_coords2):
+        return False
+
+    return all(np.allclose(c1, c2, atol=tol) for c1, c2 in zip(norm_coords1, norm_coords2))
+
+
+@pytest.fixture
+def plot_widget(qtbot):
+    """Fixture for creating a PlotWidget instance."""
+    model = DataModel()
+    widget = PlotWidget(None, model)
+    qtbot.addWidget(widget)
+    yield widget
+    widget.deleteLater()
```

Comment:
Once you have added widget to qtbot, then you should not use `deleteLater`.

Reply:
Hi @Czaki !

I do see the error:
```
E       AssertionError: Some instance of QtViewer is not properly cleaned in one of previous test. For easier debug one may use --save-leaked-object-graph flag for pytest to get graph of leaked objects. If you use qtbot (from pytest-qt) to clean Qt objects after test you may need to switch to manual clean using `deleteLater()` and `qtbot.wait(50)` later.
E       assert 1 == 0
```
when I run pytest locally. 

You are correct, the additional `deleteLater` is unnecessary. I added it following the error message as I was getting desperate. I will remove it to avoid confusion.

Now, the minimum example to reproduce the error is:
```
import pytest
from napari_spatialdata._model import DataModel
from napari_spatialdata._scatterwidgets import PlotWidget

@pytest.fixture
def plot_widget(qtbot):
    """Fixture for creating a PlotWidget instance."""
    model = DataModel()
    widget = PlotWidget(None, model)
    qtbot.addWidget(widget)
    yield widget

def test_initialization(plot_widget):
    """Test initialization of PlotWidget."""
    assert plot_widget is not None

def test_elementwidget(make_napari_viewer):
    _ = make_napari_viewer()
    assert 1 == 1
```

However, it's not the problem specific to this branch. If I remove all tests from test_scatterwidgets.py, the error persists:
![image](https://github.com/user-attachments/assets/82db6f6c-d980-463b-b515-92b6313aa6d8)

Finally, the error is triggered only once. If I run the fake test (the code pasted in above) as test_aaa.py before any other test, the error appears in the fake test but is not triggered again by any of the original tests:
![image](https://github.com/user-attachments/assets/a15bf9cd-0db4-4e9a-a0b7-e7783145f09c)

Any ideas what is going on here?

Reply:
It is not Idea. It is knowledge created in pain. 
You may also see this error in tests executed on main. 
Will try to fix this. 
In general, the problem comes from previous tests that are spawning the Viewer. 

in a general `make_napari_viewer` fixture has few weak spots (but writing a test without using it, may lead to even more spectacular failures) 

I even have opened some PR to fix it, but it is really time-consuming. 
for example https://github.com/napari/napari/pull/6745